### DISCUSSION TITLE
Backward Compatibility Check

### DISCUSSION THREAD

**Discussion by yzhao244** [serverlessworkflow/specification#474] [Ideas]
I would like to open a new discussion here regarding backward compatibility check against on-going release. 

As our community is expanding and more products/projects are adapting to CNCF serverless workflow, it may start to make sense introducing some level of backward compatibility check against our on-going release to ensure we can provide better and suitable changes to our on-going releases while product/projects which are leveraging CNCF serverless workflow can still enjoy the future releases of the specification. :) 

Please let me know what do you think. :)

I personally use json-schema-diff-validator for checking backward compatibility. And the following is screenshot after comparing workflow.json files between v0.6 and v0.7 
https://bitbucket.org/atlassian/json-schema-diff-validator/src/master/
![image](https://user-images.githubusercontent.com/15642771/132377550-a6ce0b6f-0498-4513-b6a8-835d695938b7.png)

**Comment by tsurdilo** [comment]
This is definitely what we have to do starting with the 0.7 release. From 0.7 to 1.0 we promised not to have any backwards compatible changes. For 0.7 we did remove the subflow state and made it into an action with that the "repeat" which was specific to subflow state was also removed in favor of looping transitions....

The second path is most likely a fix for the callback state json schema and is not a backwards-compatible change.

**Comment by tsurdilo** [comment]
I don't think that diff is a good tool to catch all backwards incompatible changes but certainly helps to some degree.
Would you be willing to help us set up some sort of automated way to run these types of diffs maybe on PR if this is possible?
